From 62e36af01114aba6e37736f504824dffcb5124dd Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Fri, 26 Sep 2025 09:50:28 +0200
Subject: [PATCH 217/553] Mark QUnicodeTables as security-significant (=
 default)

This component does not process external data, so it is not
security-critical.

Yes, the characters presented to its functions may come from external
sources, but that's not different from, say, a QRect being parsed from
-geometry. The fact that there is code that parses a -geometry into a
QRect doesn't make QRect a data-parser, or security-critical. It's
just container for the result, and so is QUnicodeTables: a container
for char32_t-indexed properties.

This accompanying qt_attribution.json confirms that this component is
not security-critical.

Task-number: QTBUG-135195
Pick-to: 6.8
Change-Id: I565bd885220c0282ce7fb801411f12a80052465f
Reviewed-by: Ivan Solovev <ivan.solovev@qt.io>
(cherry picked from commit f7fd1f55ccb1ae277a8daaaa6a1ee3e64bd4872b)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/text/qunicodetables.cpp | 1 +
 src/corelib/text/qunicodetables_p.h | 1 +
 util/unicode/main.cpp               | 1 +
 3 files changed, 3 insertions(+)

diff --git a/src/corelib/text/qunicodetables.cpp b/src/corelib/text/qunicodetables.cpp
index 22bc7f92874..22a665089e3 100644
--- a/src/corelib/text/qunicodetables.cpp
+++ b/src/corelib/text/qunicodetables.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: Unicode-3.0
+// Qt-Security score:significant reason:default
 
 /* This file is autogenerated from the Unicode 16.0 database. Do not edit */
 
diff --git a/src/corelib/text/qunicodetables_p.h b/src/corelib/text/qunicodetables_p.h
index d7b71f57d05..ac624dba789 100644
--- a/src/corelib/text/qunicodetables_p.h
+++ b/src/corelib/text/qunicodetables_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: Unicode-3.0
+// Qt-Security score:significant reason:default
 
 /* This file is autogenerated from the Unicode 16.0 database. Do not edit */
 
diff --git a/util/unicode/main.cpp b/util/unicode/main.cpp
index 1868d6d3841..b5bc57bde3b 100644
--- a/util/unicode/main.cpp
+++ b/util/unicode/main.cpp
@@ -3362,6 +3362,7 @@ int main(int, char **)
     QByteArray header =
         "// Copyright (C) 2020 The Qt Company Ltd.\n"
         "// SPDX-License-Identifier: Unicode-3.0\n"
+        "// Qt-Security score:significant reason:default\n"
         "\n";
 
     QByteArray note =
-- 
2.51.2

